src/transport/Session.cpp

```diff
@@ -28,24 +28,48 @@ SecureSession * Session::AsSecureSession()
     return static_cast<SecureSession *>(this);
 }
 
+const SecureSession * Session::AsSecureSession() const
+{
+    VerifyOrDie(GetSessionType() == SessionType::kSecure);
+    return static_cast<const SecureSession *>(this);
+}
+
 UnauthenticatedSession * Session::AsUnauthenticatedSession()
 {
     VerifyOrDie(GetSessionType() == SessionType::kUnauthenticated);
     return static_cast<UnauthenticatedSession *>(this);
 }
 
+const UnauthenticatedSession * Session::AsUnauthenticatedSession() const
```

Comment:
Overloads varying only on constness are confusing and don't do as we think most days. Suggest we name them AsConstXxx

Reply:
So... We could try, but then callers have to carefully think about which one they need to call depending on what they have and what they plan to call on it.  It's almost easier to just make the "get logging session id" method non-const, though that's kind of confusing too....

That said, since the only uses for these are internal I guess I could just name them whatever and make them private...